src/components/Cluster/ClusterDetail/NodePool.js

```diff
@@ -53,28 +67,13 @@ class NodePool extends Component {
   editNodePoolName = name => {
     const { cluster, nodePool } = this.props;
 
-    return new Promise((resolve, reject) => {
-      // Early return in case the name is not changed.
-      if (nodePool.name === name) return resolve();
-
-      return this.props
-        .dispatch(nodePoolActions.nodePoolPatch(cluster.id, nodePool, { name }))
-        .then(() => {
-          new FlashMessage(
-            'Succesfully edited node pool name.',
-            messageType.SUCCESS,
-            messageTTL.MEDIUM
-          );
-
-          return resolve();
-        })
-        .catch(error => {
-          // eslint-disable-next-line no-console
-          console.error(error);
-
-          return reject(error);
-        });
-    });
+    try {
+      this.props.dispatch(
+        nodePoolActions.nodePoolPatch(cluster.id, nodePool, { name })
+      );
+    } catch {
+      // Do nothing on purpose, flash message is already displayed inside the action
```

Comment:
I think if are handling errors at the action level, then we should just leave out the `try/catch` block here and remove the `throw` from the action. 

And then possibly we can do a manual post to slack using the error reporting library if we still want to be aware of cases where nodepool names fail to update.

So it becomes always the action layer's responsibility to catch any errors and not throw them any further up into the UI. 

If somehow an error does happen in the UI (like imagine maybe we this.props.dispatch is undefined for whatever reason) then I think the UI should fail, possibly with an error boundary, and we should get a notification about it.

Like this we could make a mistake (possible the compiler, linter, or tests, will catch it, but who knows...) where something is undefined or otherwise wrong on one of these lines, and we wouldn't know about it.

Reply:
Yes, that is a very valid opinion. The problem is is with how the patch functions are currently created, and how they throw errors back into the UI. I started refactoring the patch requests when I starting working on this issue, but it just seemed to be too big of a change (and a risk) for this PR (for example, the cluster patch functions are also used for scaling and upgrading clusters).

Just removing the `throw` from the actions makes some tests fail, so I'd prefer to do that refactor after I get done with this PR.

However, knowing if the feature fails is very important, so I'll add a manual error report (for now, until I refactor it) 👍